### PR TITLE
Update Run Status Message to Run Blocked during File Trust Warning

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Per https://jira.autodesk.com/browse/DYN-4998, update run status message.

After:
image

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Update Run Status Message to Run Blocked during File Trust Warning

### Reviewers

@DynamoDS/dynamo 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 1]
```diff
@@ -1605,17 +1605,12 @@ private void Open(object parameters)
...
-                    bShowFileTrustWarning = true;
-                }
                 ExecuteCommand(new DynamoModel.OpenFileCommand(filePath, forceManualMode));
-
-                if (FileTrustViewModel != null && bShowFileTrustWarning == true)
+                if (!PreferenceSettings.TrustedLocations.Contains(directoryName) 
+                    && DynamoModel.IsTestMode == false
+                    && FileTrustViewModel != null)
                 {
+                    HomeSpace.RunSettings.ForceAutomaticWithoutRun = true;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:1613]: @RobertGlobant20 I included a minimal fix for now and plan to rethink about this boolean with you a bit to cover the periodic run mode as well

File: src/DynamoCore/Models/RunSettings.cs [hunk 2]
```diff
@@ -20,8 +20,10 @@ public class RunSettings : NotificationObject
         private RunType runType;
         private bool runEnabled;
         private bool runTypesEnabled;
-        private bool runTypesComboBoxToolTipIsEnabled;
 
+        /// <summary>
+        /// Boolean indicates if Dynamo should block run even mode is Automatic.
```
Comment (zeusongit) [inline src/DynamoCore/Models/RunSettings.cs:25]: Should be updated to: `Boolean indicates if Dynamo should block run even when mode is Automatic.`

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 3]
```diff
@@ -1605,17 +1605,12 @@ private void Open(object parameters)
                 }
 
                 var directoryName = Path.GetDirectoryName(filePath);
-                //Checks if the file that is being opened is in the trusted list.
-                bool bShowFileTrustWarning = false;
-                if(!PreferenceSettings.TrustedLocations.Contains(directoryName) && DynamoModel.IsTestMode == false)
-                {
-                    HomeSpace.RunSettings.ForceAutomaticWithoutRun = true;
-                    bShowFileTrustWarning = true;
```
Comment (zeusongit) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:1613]: So we do not need to set `bShowFileTrustWarning` true anymore?

**Review by RobertGlobant20 — APPROVED** [review]
LGTM just consider that if ForceAutomaticWithoutRun is changed to be static, then some parts will need to be updated.

File: src/DynamoCore/Models/RunSettings.cs (hunk 2 above)
Comment (QilongTang) [reply src/DynamoCore/Models/RunSettings.cs:25]: Will do

File: src/DynamoCoreWpf/ViewModels/Core/HomeWorkspaceViewModel.cs [hunk 4]
```diff
@@ -265,6 +265,12 @@ void hwm_EvaluationCompleted(object sender, EvaluationCompletedEventArgs e)
         /// </summary>
         internal void UpdateRunStatusMsgBasedOnStates()
         {
+            if ((bool)((Model as HomeWorkspaceModel)?.RunSettings.ForceAutomaticWithoutRun))
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Core/HomeWorkspaceViewModel.cs:268]: won't this cast fail if Model is a customWorkspaceModel? Or will null cast to false?

Comment (QilongTang) [reply src/DynamoCoreWpf/ViewModels/Core/HomeWorkspaceViewModel.cs:268]: @mjkkirschner I did confirm that this function is bind to graph evaluation so the above case would not happen

**Comment by QilongTang** [comment]
@RobertGlobant20 @zeusongit I am going to merge this so the code is synced between us when we work on all following improvements

**Pull request merged**